d2core/d2gui/label.go

```diff
@@ -39,3 +33,25 @@ func (l *Label) render(target d2render.Surface) error {
 func (l *Label) getSize() (int, int) {
 	return l.surface.GetSize()
 }
+
+func (l *Label) GetText() string {
+	return l.text
+}
+
+func (l *Label) SetText(text string) error {
+	return l.setText(text)
+}
+
+func (l *Label) setText(text string) error {
+	width, height := l.font.GetTextMetrics(text)
```

Comment:
This needs to simply return if they text has not changed. No reason to re-generate the graphics if the change is a no-op.

Reply:
good catch 👍